raphtory/src/graph_loader/example/sx_superuser_graph.rs

```diff
@@ -75,20 +76,30 @@ pub fn sx_superuser_file() -> Result<PathBuf, Box<dyn std::error::Error>> {
 
 /// Load the SX SuperUser dataset into a graph and return it
 ///
-/// # Arguments
-///
-/// * `shards` - The number of shards to use for the graph
-///
 /// Returns:
 ///
 /// - A Result containing the graph or an error
-pub fn sx_superuser_graph() -> Result<Graph, Box<dyn std::error::Error>> {
+pub fn sx_superuser_graph(num_layers: Option<usize>) -> Result<Graph, Box<dyn std::error::Error>> {
     let graph = Graph::new();
     CsvLoader::new(sx_superuser_file()?)
         .set_delimiter(" ")
         .load_into_graph(&graph, |edge: TEdge, g: &Graph| {
-            g.add_edge(edge.time, edge.src_id, edge.dst_id, NO_PROPS, None)
+            if let Some(layer) = num_layers
+                .map(|num_layers| calculate_hash(&(edge.src_id, edge.dst_id)) % num_layers as u64)
+                .map(|id| id.to_string())
+            {
+                g.add_edge(
+                    edge.time,
+                    edge.src_id,
+                    edge.dst_id,
+                    NO_PROPS,
+                    Some(layer.as_str()),
+                )
                 .expect("Error: Unable to add edge");
+            } else {
+                g.add_edge(edge.time, edge.src_id, edge.dst_id, NO_PROPS, None)
+                    .expect("Error: Unable to add edge");
+            }
```

Comment:
Can we make this a separate function in the benchmark instead? It is a really strange parameter to have for this dataset